src/Modal/Modal.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from "react";
+import "./modal-style.scss";
+
+interface ModalProps {
+    toggle: boolean;
+    id?: string;
+    fullscreen?: boolean;
+    position?: "left" | "right";
+    className?: string;
+    disableBackdropDismiss?: boolean;
+    ariaLabel?: string;
+    ariaDescribedby?: string;
+    dismissModal?: () => void;
+    header?: React.ReactNode;
+    body?: React.ReactNode;
+    footer?: React.ReactNode;
+}
+
+interface ModalState {
+    open: boolean;
+    close: boolean;
+}
+
+export class Modal extends React.Component<ModalProps, ModalState> {
+    constructor(props: ModalProps) {
+        super(props);
+        this.state = {
+            open: false,
+            close: false
+        };
+    }
+
+    componentDidMount(): void {
+        if (this.props.toggle === true) {
+            this.setState({ open: true });
+        }
+    }
+
+    componentDidUpdate(prevProps: ModalProps): void {
+        if (this.props.toggle !== prevProps.toggle) {
+            this.setState({ open: !prevProps.toggle, close: prevProps.toggle });
+        }
+    }
+
+    /**
+     * set modal class names
+     * @returns string of concatenated class names
+     */
+    setModalClasses(): string {
+        const classes: string = this.classNames({
+            "show": this.state.open,
+            "fade": this.state.close,
+            "modal-aside": !!this.props.position,
+            "modal-aside-left": this.props.position === "left",
+            "modal-aside-right": this.props.position === "right",
+            "modal-fullscreen": this.props.fullscreen,
+            [this.props.className]: !!this.props.className
+        });
+        return classes;
+    }
+
+    /**
+     * Convert object to string
+     * @param { [key: string]: boolean } classes list of classes to be appended
+     * @returns {string} string of concatenated class names
+     */
+    classNames(classes: { [key: string]: boolean }): string {
+        return Object.entries(classes)
+            .filter(([key, value]) => value)
+            .map(([key, value]) => key)
+            .join(" ");
+    }
+
+    /**
+     * emit dismissModal event when backdrop is clicked
+     * @param {React.MouseEvent} event clicked element
+     */
+    closeModal(event: React.MouseEvent): void {
+        const target: HTMLElement = event.target as HTMLElement;
+        if (event && event.target && target.classList && target.classList.length) {
+            const classList: DOMTokenList = target.classList;
+            if (classList.contains("modal")) {
+                this.setState({ open: false, close: true });
+                this.props.dismissModal();
+            }
+        }
+    }
+
+    render() {
+        return (
+            <div
+                className={"modal-backdrop" + (this.state.open ? " show" : " fade")}
+                onClick={(event: React.MouseEvent) => { !this.props.disableBackdropDismiss && this.closeModal(event); }}
```

Comment:
no need to put the logic here in the template, just use this instead (with renaming it):
```jsx
onClick={this.onDismiss}
```
make sure to bind the method in the `constructor`
```typescript
this.onDismiss = this.onDismiss.bind(this);
```
and move the logic to the `onDismiss` method

Reply:
onDismiss is a prop it shouldn't trigger if the user choose to disable the back drop dismiss. So we need to keep the condition
and the event should not trigger only when the user clicks in the backdrop, your suggestion will not handle those cases.

Reply:
The method that I'm suggesting `this.onDismiss` is different from `this.props.onDismiss` that's why I was asking you to bind it in the constructor since it's a local method that hosts whatever logic is related to dismissing the modal. You already have such a method, it's called `closeModal`. So my previous comment was to suggest that you rename the method and move this logic inside it. You could keep the same name but at least change the `JSDoc` since it says that it will **emit** a dismiss modal, I guess this comment was copied directly from its Angular counterpart.